pandas/core/groupby.py

```diff
@@ -2626,6 +2626,14 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     elif isinstance(key, BaseGrouper):
         return key, [], obj
 
+    if key is None and isinstance(group_axis, MultiIndex):
```

Comment:
can you simplify this. e.g. maybe put this coercion higher (before the giant if/then)